lib/authentication/view/sign_up_view.dart

```diff
@@ -45,25 +51,46 @@ class _SignUpViewState extends State<SignUpView> {
     _fetchFilter();
   }
 
+  /// This function takes the email of user and sends the name data to the name fields.
+  /// Special characters such as ".", "_" are used to separate the names, and numbers are removed.
+  /// Format example: firstnameone_firstnametwo.lastname123@stud.acs.pub.ro
+  /// The names will start with capital letters.
+
+  void parseNameFromEmail(TextEditingController email,
+      TextEditingController firstName, TextEditingController lastName) {
+    String emailWithoutNumbers = email.text.replaceAll(RegExp(r'[0-9]'), '');
+    List<String> names = emailWithoutNumbers.split('.');
+    List<String> firstNames;
+
+    if (!names[0].contains('_')) {
+      firstName.text = names[0].titleCase;
+    } else {
+      firstNames = names[0].split('_');
```

Comment:
Declare firstNames here instead of above. It should work fine, and it's usually better to declare variables as close to the place where they are used as possible.